Utils/AuthenticationKeychainInfo.swift

```diff
@@ -84,6 +88,13 @@ public class AuthenticationKeychainInfo: NSObject, NSCoding {
 
 // MARK: - API
 public extension AuthenticationKeychainInfo {
+    enum AuthenticationPurpose {
```

Comment:
Nit: No need to declare this in the `AuthenticationKeychainInfo` scope. I would pull this out.